File: app/src/main/java/com/b18/kipsafe/alarms/AlarmScheduler.java
package com.b18.kipsafe.alarms;

import android.app.AlarmManager;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;

import java.util.Calendar;

/**
 * Schedule alarms.
 */
class AlarmScheduler {

    private Context context;

    AlarmScheduler(Context context) {
        this.context = context;
    }

    /**
     * Schedule Android alarm, now by default daily.
     *
     * @param timeCal Time to schedule alarm for.
     */
    void scheduleAlarm(Calendar timeCal) {
        //schedule alarm
        //time received is in UTC
        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
        alarmManager.setExact(AlarmManager.RTC_WAKEUP, timeCal.getTimeInMillis(), getPendingIntent());
    }

    /**
     * Cancel the alarm that was set.
     */
    void cancelAlarm() {
        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
        alarmManager.cancel(getPendingIntent());

    }

    private PendingIntent getPendingIntent() {
        Intent intentAlarm = new Intent(context, AlarmReceiver.class);
        return PendingIntent.getBroadcast(
                context, 1, intentAlarm, PendingIntent.FLAG_UPDATE_CURRENT);
    }
}

File: app/src/main/java/com/b18/kipsafe/alarms/AlarmScheduler.kt
package com.b18.kipsafe.alarms

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import java.util.*

/**
 * Schedules alarms.
 */
class AlarmScheduler(private val context: Context?) {
    /**
     * Schedule a daily alarm.
     *
     * @param timeCal Time to schedule the alarm for, as Calendar object.
     */
    fun scheduleAlarm(timeCal: Calendar) {
        val alarmManager = context?.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        alarmManager.setExact(AlarmManager.RTC_WAKEUP, timeCal.timeInMillis, getPendingIntent())
    }

    /**
     * Cancel an alarm that has been set.
     */
    fun cancelAlarm() {
        val alarmManager = context?.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        alarmManager.cancel(getPendingIntent())
    }

    private fun getPendingIntent() : PendingIntent {
        val intentAlarm = Intent(context, AlarmReceiver::class.java)
        return PendingIntent.getBroadcast(
                context, 1, intentAlarm, PendingIntent.FLAG_UPDATE_CURRENT)
    }
}